ehrapy/tools/nlp/_medcat.py

```diff
@@ -156,213 +46,313 @@ def update_cat_config(self, concept_db_config: Config) -> None:
+                   #
+                   # 1 0 (first entitiy extracted from first row)   diabetes11  fb11 ...
+                   # 2 0 (second entitiy extracted from first row)  diabetes12  fb12 ...
+                   # 3 1 (first entitiy extracted from second row)  diabetes21  fb21 ...
+                   # 4 1 (second entitiy extracted from second row) diabetes22  fb22 ...
+
+    Args:
+        ep_cat: Ehrapy's custom MedCAT object. The annotated_results attribute will be set here.
+        obs: AnnData obs containing the free text column
+        text_column: Name of the column that should be annotated
+        n_proc: Number of processors to use
+        batch_size_chars: batch size to control for the variability between document sizes
+
+    """
+    non_null_text = _filter_null_values(obs, text_column)
+    formatted_text_column = _format_df_column(non_null_text, text_column)
+    results = ep_cat.cat.multiprocessing(formatted_text_column, batch_size_chars=batch_size_chars, nproc=n_proc)
+    flattened_res = _flatten_annotated_results(results)
+    # sort for row number in ascending order and reset index to keep index updated
+    ep_cat.annotated_results = _annotated_results_to_df(flattened_res).sort_values(by=["row_nr"]).reset_index(drop=True)
+
+
+def get_annotation_overview(ep_cat: MedCAT, n: int = 10, status: str = "Affirmed", save_to_csv: bool = False) -> None:
+    """Provide an overview for the annotation results. An overview will look like the following:
+       cui (the CUI), nsubjects (from how many rows this one got extracted), type_ids (TUIs), name(name of the entitiy), perc_subjects (how many rows relative
+       to absolute number of rows)
+
+    Args:
+        ep_cat: The current MedCAT object which holds all infos on medcat analysis with ehrapy.
+        n: Basically the parameter for head() of pandas Dataframe. How many of the most common entities should be shown?
+        status: One of "Affirmed" (default), "Other" or "Both". Displays stats for either only affirmed entities, negated ones or both.
+        save_to_csv: Whether to save the overview dataframe to a local .csv file in the current working directory or not.
+
+    Returns:
+        A pandas DataFrame with the overview stats.
+    """
+    df = _filter_df_by_status(ep_cat.annotated_results, status)
+    # group by CUI as this is a unique identifier per entity
+    grouped = df.groupby("cui")
+    # get absolute number of rows with this entity
+    # note for overview, only one TUI and type is shown (there shouldn't be much situations were multiple are even possible or useful)
+    res = grouped.agg(
+        {
+            "pretty_name": (lambda x: next(iter(set(x)))),
+            "type_ids": (lambda x: next(iter(x))[0]),
+            "types": (lambda x: next(iter(x))[0]),
+            "row_nr": "nunique",
+        }
+    )
+    res = res.rename(columns={"row_nr": "n_patients"})
```

Comment:
Think that by default we take a patient_visit view. Might be a better naming here.